pkg/keccak/Makefile

```diff
@@ -0,0 +1,24 @@
+PKG_NAME    := keccak
+PKG_URL     := https://github.com/gvanas/KeccakCodePackage
+PKG_VERSION := bc592d1780a6b3204c26c6f872552ace8fc6fd65
+
+ifeq ($(CPU),native)
+  CPU_ARCHIVE_NAME := Native
+else
+ifeq ($(CPU_ARCH),$(filter $(CPU_ARCH), cortex-m0 cortex-m0plus))
+  CPU_ARCHIVE_NAME := ARMv6M
+else
+ifeq ($(CPU_ARCH),$(filter $(CPU_ARCH), cortex-m3 cortex-m4 cortex-m4f cortex-m7 cortex-m7f))
+  CPU_ARCHIVE_NAME := ARMv7M
+endif
+endif
+endif
+
+.PHONY: all
+
+all:
```

Comment:
General note on this Makefile: it should be reworked in a way that doesn't rebuild the package sources when they are already there. This is the usual way of writing Makefile.

Reply:
```Makefile
all: $(BINDIR)/keccak.a

$(BINDIR)/keccak.a: $(PKG_BUILD_DIR)/bin/KeccakHash$(CPU_ARCHIVE_NAME).a

$(PKG_BUILD_DIR)/bin/KeccakHash$(CPU_ARCHIVE_NAME).a
	@cp -r $(RIOTBASE)/pkg/keccak/contrib/Modes $(PKG_BUILD_DIR)
	$(MAKE) KeccakHash$(CPU_ARCHIVE_NAME).a -C $(PKG_BUILD_DIR)
	@cp $(PKG_BUILD_DIR)/bin/KeccakHash$(CPU_ARCHIVE_NAME).a $(BINDIR)/keccak.a
```

Like so?

Reply:
Maybe better like this:

```Makefile
all: $(BINDIR)/keccak.a

$(BINDIR)/keccak.a: $(PKG_BUILD_DIR)/bin/KeccakHash$(CPU_ARCHIVE_NAME).a
      	@cp $(PKG_BUILD_DIR)/bin/KeccakHash$(CPU_ARCHIVE_NAME).a $(BINDIR)/keccak.a

$(PKG_BUILD_DIR)/bin/KeccakHash$(CPU_ARCHIVE_NAME).a
	@cp -r $(RIOTBASE)/pkg/keccak/contrib/Modes $(PKG_BUILD_DIR)
	$(MAKE) KeccakHash$(CPU_ARCHIVE_NAME).a -C $(PKG_BUILD_DIR)
```

Reply:
Alright, I will use that in the next commit.